docs/add-ons/argocd.md

```diff
@@ -49,3 +49,30 @@ argocd_applications = {
   }
 }
 ```
+
+### Add-ons 
+
+A common operational pattern is to leverage Infrastructure as Code for provisioning EKS clusters (in addition to other AWS resources) and GitOps for managing cluster configuration. The framework provides support for this approach by leveraging the ArgoCD [App of Apps](https://argo-cd.readthedocs.io/en/stable/operator-manual/cluster-bootstrapping/) pattern. 
+
+To configure the framework to leverage ArgoCD for managing add-ons, you must pass configuration for a root ArgoCD Application that points to your desired add-ons. You can specify the root application by setting the `add_on_application` value to true in your application configuration.  
+
+Additionally, you must set the `argocd_manage_add_ons` property to true. When this flag is set, the framework will still provision all AWS resources necessary to support add-on functionality, but it will not apply Helm charts directly via Terraform. Instead, the framework will pass AWS resource values needed for each add-on to ArgoCD via the values map of the root add-on Application. For specific values passed for each add-on, see the individual add-on documentation. 
+
+Sample configuration can be found below: 
+
+```
+argocd_enable           = true
+argocd_manage_add_ons   = true
+argocd_applications     = {
+  infra = {
+    namespace             = "argocd"
+    path                  = "<path>"
+    repo_url              = "<repo_url>"
+    target_revision       = "HEAD"
+    destination           = "https://kubernetes.default.svc"
+    project               = "default"
+    values                = {}
+    add_on_application    = true # Indicates the root add-on application. 
+  }
+}
+```
```

Comment:
Maybe just call out that alternatively if customers wish to do a CD pipeline (like GitLab) that's possible by using the default behavior.

Reply:
Agree - although I plan to add this content to a top level `add-ons` doc that discusses the different approaches.